ddtrace/_trace/provider.py

```diff
@@ -14,9 +14,31 @@
 log = get_logger(__name__)
 
 
-_DD_CONTEXTVAR: contextvars.ContextVar[Optional[Union[Context, Span]]] = contextvars.ContextVar(
-    "datadog_contextvar", default=None
-)
+class ContextVarManager:
+    """
+    In the implementation of ContextVar, when a key is re-associated with a new value the underlying HAMT must clone
+    a level of the tree in order to maintain immutability. This operation requires de-referencing pointers to Python
+    objects stored in the Context, which typically includes objects not created or managed by this library. It's
+    possible for such objects to have mis-managed reference counts (speculatively:  in order to convert their
+    ContextVar storage from a strong to a weak reference. When such objects are de-referenced--as they would be when
+    a reassoc from this code forces a clone--it could cause heap corruption or a segmentation fault.
+
+    Accordingly, we try to prevent reassoc events when possible by storing a long-lived wrapper object and only setting
+    the target value within that object.
+    """
+
+    def __init__(self, name: str):
+        self._context_var = contextvars.ContextVar[Optional[Union[Context, Span]]](name, default=None)
+
+    def get(self) -> Optional[Union[Context, Span]]:
+        return self._context_var.get()
+
+    def set(self, value: Optional[Union[Context, Span]]) -> None:
+        self._context_var.set(value)
```

Comment:
It seems like we removed the wrapper along the way ?
The aim was to have `self._context_var.get().wrapper_set(value)`
I am not sure why we got rid of it ?